transcom/mymove
# INT-B-21478 Add Oconus Rate area to prime v3 getMoveTaskOrder  api

## [B-21478](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1016131&RoomContext=TeamRoom%3A855664&concept=TeamRoom)

## Summary

This PR introduces changes to the Prime V3 getMoveTaskOrder API to return Oconus (ie.. Alaska postal codes) Rate Area for all shipments.

It introduces two new OPTIONAL fields: **originRateArea, destinationRateArea**:

![Screenshot 2024-11-05 at 7 34 34 AM](https://github.com/user-attachments/assets/4f699025-87d8-46c6-9459-196bbf0b2e84)
![Screenshot 2024-11-05 at 7 34 17 AM](https://github.com/user-attachments/assets/0cbd0900-0416-45b6-9c92-d7eb9859695d)

`select * from re_rate_areas where id = '<ID>';`

RateArea display is derived off Pickup and Destination postal codes. originRateArea/destinationRateArea will only be present if code is an **OCONUS**(Alaska) zip code and its location is determined based on shipment type. All non-PPM shipment will display rateArea on the root level of shipment while PPM will display on the PPMShipment level.


[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Login as customer and create and submit a move containing all shipment types. For each shipment use all permutations for pickup and destination addresses with alaska and other continental states. **NOTE -- at this time the system is not capable of accepting Alaska postal codes for PPM shipments. Need to follow up with Danny M on how to test with current restriction.** For non-PPMs, you can use Alaska postal codes but you can't use AK for state. For now, just select any state but make sure to use an Alaska postal code.  `select distinct state, postal_code from addresses where state = 'AK'; `
2. Login as service counselor and click <submit move details> for move.
3. Login as TOO and find move. Select all shipments and click <Approve Selected>
4. Login Prime Sim. Open web developer and go to network tab. Click on move and go to details view.  In network tab view, select getMoveTaskOrder request. It will typically be at the bottom of the network request stack. View response json and confirm originRateArea/destinationRateArea if present/not present based on test cases.
5. Verify placement of rateArea information. Root level of shipment for non-PPM shipments and PPMShipment level for PPMs.
![Screenshot 2024-11-05 at 8 30 29 AM](https://github.com/user-attachments/assets/3d72ce8c-0ce4-4860-b60d-b8fcbe8a4896)
![Screenshot 2024-11-05 at 7 18 19 AM](https://github.com/user-attachments/assets/8d2320ab-f7f3-4acf-801a-6c2b7832e521)
6. Goto http://primelocal:3000/swagger-ui/prime_v3.html#/moveTaskOrder/getMoveTaskOrder. Copy moveID from Prime Sim UI and execute. Verify rateArea info. Must match what is seen in step 5.


### Frontend


### Backend
